File: src/main/java/com/stheren/email_reader/engine/data/Email.java
package com.stheren.email_reader.engine.data;

import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.UUID;

import com.fasterxml.jackson.annotation.JsonAnyGetter;
import com.fasterxml.jackson.annotation.JsonAnySetter;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyDescription;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;

import javax.annotation.processing.Generated;

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonPropertyOrder({
    "_id",
    "metadata",
    "content",
    "history",
    "extentions"
})
@Generated("jsonschema2pojo")
public class Email {

    /**
     * uid RFC4122
     * (Required)
     * 
     */
    @JsonProperty("_id")
    @JsonPropertyDescription("uid RFC4122")
    private UUID id;
    /**
     * 
     * (Required)
     * 
     */
    @JsonProperty("metadata")
    private Metadata metadata;
    /**
     * 
     * (Required)
     * 
     */
    @JsonProperty("content")
    private String content;
    @JsonProperty("history")
    private UUID history;
    @JsonProperty("extentions")
    private List<Extention> extentions = null;
    @JsonIgnore
    private Map<String, Object> additionalProperties = new HashMap<String, Object>();

    /**
     * No args constructor for use in serialization
     * 
     */
    public Email() {
    }

    /**
     * 
     * @param extentions
     * @param metadata
     * @param id
     * @param history
     * @param content
     */
    public Email(UUID id, Metadata metadata, String content, UUID history, List<Extention> extentions) {
        super();
        this.id = id;
        this.metadata = metadata;
        this.content = content;
        this.history = history;
        this.extentions = extentions;
    }

    /**
     * uid RFC4122
     * (Required)
     * 
     */
    @JsonProperty("_id")
    public UUID getId() {
        return id;
    }

    /**
     * uid RFC4122
     * (Required)
     * 
     */
    @JsonProperty("_id")
    public void setId(UUID id) {
        this.id = id;
    }

    /**
     * 
     * (Required)
     * 
     */
    @JsonProperty("metadata")
    public Metadata getMetadata() {
        return metadata;
    }

    /**
     * 
     * (Required)
     * 
     */
    @JsonProperty("metadata")
    public void setMetadata(Metadata metadata) {
        this.metadata = metadata;
    }

    /**
     * 
     * (Required)
     * 
     */
    @JsonProperty("content")
    public String getContent() {
        return content;
    }

    /**
     * 
     * (Required)
     * 
     */
    @JsonProperty("content")
    public void setContent(String content) {
        this.content = content;
    }

    @JsonProperty("history")
    public UUID getHistory() {
        return history;
    }

    @JsonProperty("history")
    public void setHistory(UUID history) {
        this.history = history;
    }

    @JsonProperty("extentions")
    public List<Extention> getExtentions() {
        return extentions;
    }

    @JsonProperty("extentions")
    public void setExtentions(List<Extention> extentions) {
        this.extentions = extentions;
    }

    @JsonAnyGetter
    public Map<String, Object> getAdditionalProperties() {
        return this.additionalProperties;
    }

    @JsonAnySetter
    public void setAdditionalProperty(String name, Object value) {
        this.additionalProperties.put(name, value);
    }

}

File: src/main/kotlin/com/stheren/email_reader/data/Email.kt
package com.stheren.email_reader.data

import com.fasterxml.jackson.annotation.*
import java.util.*
import javax.annotation.processing.Generated

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonPropertyOrder("_id", "metadata", "content", "history", "extentions")
class Email {

    @get:JsonProperty("_id")
    @set:JsonProperty("_id")
    @JsonProperty("_id")
    @JsonPropertyDescription("uid RFC4122")
    lateinit var id: UUID

    @get:JsonProperty("metadata")
    @set:JsonProperty("metadata")
    @JsonProperty("metadata")
    lateinit var  metadata: Metadata

    @get:JsonProperty("content")
    @set:JsonProperty("content")
    @JsonProperty("content")
    lateinit var  content: String

    @get:JsonProperty("history")
    @set:JsonProperty("history")
    @JsonProperty("history")
    var history: UUID? = null

    @get:JsonProperty("extentions")
    @set:JsonProperty("extentions")
    @JsonProperty("extentions")
    var  extentions: List<Extention>? = null

    @JsonIgnore
    private val additionalProperties: MutableMap<String, Any> = HashMap()

    constructor() {}

    constructor(
        id: UUID,
        metadata: Metadata,
        content: String,
        history: UUID,
        extentions: List<Extention>
    ) : super() {
        this.id = id
        this.metadata = metadata
        this.content = content
        this.history = history
        this.extentions = extentions
    }

    @JsonAnyGetter
    fun getAdditionalProperties(): Map<String, Any> {
        return additionalProperties
    }

    @JsonAnySetter
    fun setAdditionalProperty(name: String, value: Any) {
        additionalProperties[name] = value
    }
}